### ISSUE TITLE
Issues updating to nakama 3.4.0

### ISSUE THREAD

**Issue by ernesernesto** [heroiclabs/nakama#643]
Hi, we're updating from nakama 3.1.2 into nakama 3.4.0 and it seems like we've encountered a regression bug on tournaments/leaderboard

```
github.com/heroiclabs/nakama/v3/internal/cronexpr.(*Expression).Next(0x0, 0x2360ad1b, 0xed87dfdec, 0x0, 0x7f2f521363b0, 0x2360ad1b, 0xed87dfdec)
github.com/heroiclabs/nakama/v3/internal/cronexpr/cronexpr.go:174 +0xad

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x12516cd]

goroutine 251 [running]:
github.com/heroiclabs/nakama/v3/internal/cronexpr.(*Expression).Next(0x0, 0x2360ad1b, 0xed87dfdec, 0x0, 0x7f2f521363b0, 0x2360ad1b, 0xed87dfdec)
	github.com/heroiclabs/nakama/v3/internal/cronexpr/cronexpr.go:174 +0xad
github.com/heroiclabs/nakama/v3/internal/cronexpr.(*Expression).NextN(0x0, 0x2360ad1b, 0xed87dfdec, 0x0, 0x2, 0xc000d65318, 0x0, 0xc000d65858)
```

We're usually create leaderboard with this cron expression */5 * * * * or 0 2 * * 1

Additionally, it looks like client sdk that we use is not compatible with 3.4.0 releases? We got this error when trying to connect to nakama 

```
System.IO.EndOfStreamException: Unexpected end of stream encountered whilst attempting to read 2 bytes
  at Nakama.Ninja.WebSockets.Internal.BinaryReaderWriter+<ReadExactly>d__0.MoveNext () [0x00118] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.Ninja.WebSockets.Internal.WebSocketFrameReader+<ReadAsync>d__0.MoveNext () [0x0009b] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.Ninja.WebSockets.Internal.WebSocketImplementation+<ReceiveAsync>d__33.MoveNext () [0x000e7] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.Ninja.WebSockets.Internal.WebSocketImplementation+<ReceiveAsync>d__33.MoveNext () [0x0034a] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.Ninja.WebSockets.Internal.WebSocketImplementation+<ReceiveAsync>d__33.MoveNext () [0x00841] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.WebSocketAdapter+<ReceiveLoop>d__35.MoveNext () [0x0009f] in <086f3f08159042feac393a5f976be30a>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Nakama.WebSocketAdapter+<Connect>d__31.MoveNext () [0x001a5] in <086f3f08159042feac393a5f976be30a>:0 
UnityEngine.Debug:LogError(Object)
Nakama.Socket:<.ctor>b__51_2(Exception)
Nakama.<Connect>d__31:MoveNext()
UnityEngine.UnitySynchronizationContext:ExecuteTasks() (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:107)
```

Nakama: 3.1.2 -> 3.4.0
client sdk: nakama 2.7.0 unity

Thanks!

**Comment by novabyte** [comment]
@ernesernesto I suspect that this issue might be related to the timezone of the server rather than a regression. What timezone do you run the game server on?

**Comment by ernesernesto** [comment]
@novabyte here is the output of timedatectl on the server 

```
timedatectl
      Local time: Tue 2021-07-13 07:48:45 UTC
  Universal time: Tue 2021-07-13 07:48:45 UTC
        RTC time: Tue 2021-07-13 07:48:45
       Time zone: Etc/UTC (UTC, +0000)
 Network time on: yes
NTP synchronized: yes
 RTC in local TZ: no
```

We're using Nakama 2.7.0 Unity sdk and Unity 2019.4.18f1

```
Can you share the specific client SDK code which causes that exception to be thrown and what version of Unity engine you're on?
```
Will need to dug into this since we ended up downgrading the server back and not triggering on 3.1.2 nakama

**Comment by zyro** [comment]
@ernesernesto This is resolved in d94a708115c3a899439236bd02c60965e20b3743 and will be part of the next release.
